server/src/main/java/org/loomdev/loom/scheduler/SchedulerImpl.java

```diff
@@ -1,20 +1,18 @@
 package org.loomdev.loom.scheduler;
 
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 import org.loomdev.api.plugin.PluginManager;
 import org.loomdev.api.scheduler.Scheduler;
 import org.loomdev.api.scheduler.ScheduledTask;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.Optional;
 import java.util.concurrent.*;
-import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class SchedulerImpl implements Scheduler {
 
     private final ExecutorService asyncExecutor;
-    private final ConcurrentHashMap<Integer, ScheduledTaskImpl> taskQueue;
+    private final ConcurrentHashMap<Integer, ScheduledTask> taskQueue;
```

Comment:
This OK currently, but might need to change again later if we need access to the implementation object. I would rather case when returning then case to the implementation type, since an interface can have multiple implantations.